src/app/components/views/homepage/homepage.component.ts

```diff
@@ -13,43 +14,82 @@ import { WeatherService } from 'src/app/services/weather.service';
   private ngUnsubscribe: Subject<void> = new Subject<void>();
-  public jobList: JobData[];
   
   constructor(private weatherService: WeatherService, private jobService: JobService) { }
 
   ngOnInit(): void {
-    this.jobList = this.jobService.getJobs();
-    this.callAndSubscribeToWeatherService();
+    // this.callAndSubscribeToWeatherService(); // To be reinstated as a future version.
+    
+    this.techUUID = 'cj692r'; // TODO - Make part of a sort of "login" feature. Aaron is working on this I believe, possibly a sort of modal.
+    this.callJobServiceJobs(this.techUUID);
+    this.subscribeToJobServiceJobs();
   }
 
   ngOnDestroy(): void {
     this.ngUnsubscribe.next();
-    this.ngUnsubscribe.complete
+    this.ngUnsubscribe.complete();
   }
 
   /**
+   * @todo This method requires implementation, now that the job generation has been moved to the back end. Will require back end changes as well, requesting the back end to generate a new job. This should be done by creating a system where the front end can do: backendURL/jobs/get/{uuid}/{job-index}. Usually, when removing job-index, the back end will decide how many jobs should be returned. However, when passing job-index, the back end will then then generate that many jobs and remove them. So essentially the front end will request job-index as current job count + 1
    * @description For use in html when the request job button is clicked. Will Generate a new job to add to the job list
    * @returns {void}
    */
   public onRequestJobButtonClick(): void {
-    this.jobService.generateNewJob();
+    console.log('TODO: awaiting implementation as a future feature.');
   }
 
   /**
-   * @description - Subscribes to weather service and sets global variable for the api response
+   * @description For use in html when the refresh button is clicked. Will call JobService 
    * @returns {void}
    */
-  private callAndSubscribeToWeatherService(): void {
-    if (Array.isArray(this.jobList) && this.jobList.length > 0) { // only want to make this call if there are 
-      this.weatherService.call(this.jobList[0].lat, this.jobList[0].long); // calls with first assigned job cause alerts should be similar to the area
-      this.weatherService.getLoading().pipe(take(3), takeUntil(this.ngUnsubscribe)).subscribe({
-        next: (loading: boolean) => {
-          if (!loading && this.weatherService.hasSuccessfullyCompleted()) {
-            this.weatherAlertResponse = this.weatherService.getResults();
-          }
+  public onRefreshJobList(): void {
+    this.callJobServiceJobs(this.techUUID);
+  }
+
+  /**
+   * @description To be used in ngOnInit, will do subscribe to job service jobs so that when the api is called it will update the jobResponse data.
+   * @returns {void}
+   */
+  private subscribeToJobServiceJobs(): void {
+    // Should we take(3) here? It is set up to be able to be re-called via the refresh button. So we should take until ngOnDestroy, as it is set now?
```

Comment:
Let's do take(3)

Reply:
I decided to do take(3) because we will only ever call the job list from the homepage. So I made sure the subscription and call method happened in the same wrapper method so they would always happen together.

I do have a concern, could this "double subscribe" if you call the wrapper method before the response from the api? how best to avoid this?

Reply:
wrapped in a jobService.hasSuccessfullyCompleted(), so that it won't call the api if it's already calling!

Reply:
sorry, no hasSuccessfullyCompleted, as that has to do with api success, not api activity.

Reply:
instead I made it so that before subscribing to the api, if there is already a call in place than the old subscription is destroyed and a new one created.